core/src/main/scala/org/apache/spark/deploy/security/HadoopDelegationTokenProvider.scala

```diff
@@ -43,11 +44,13 @@ private[spark] trait HadoopDelegationTokenProvider {
    * Obtain delegation tokens for this service and get the time of the next renewal.
    * @param hadoopConf Configuration of current Hadoop Compatible system.
    * @param creds Credentials to add tokens and security keys to.
+   * @param fileSystems List of file systems for which to obtain delegation tokens.
    * @return If the returned tokens are renewable and can be renewed, return the time of the next
    *         renewal, otherwise None should be returned.
    */
   def obtainDelegationTokens(
     hadoopConf: Configuration,
     sparkConf: SparkConf,
+    fileSystems: Set[FileSystem],
```

Comment:
This is a bit odd, and ultimately I think that `YarnSparkHadoopUtil.hadoopFSsToAccess` should be used for everybody (a.k.a. moved to `HadoopFSDelegationTokenProvider`), but that should be a separate change.

Reply:
Felt the same when changed but didn't have good solution. Good suggestion, will file a jira and resolve it.